From 6c96bdf50a2a5da1dfcf22b894274970642e92a1 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Thu, 4 Jul 2019 17:19:32 +0900
Subject: [PATCH] ipts: fix crash caused by ipts_send_feedback()

ipts_send_feedback() causes touch inputs to crash on some
platforms, especially on Surface Pro 4 and Surface Book 1.
Let's comment out these lines for now.
---
 drivers/misc/ipts/ipts-hid.c | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/drivers/misc/ipts/ipts-hid.c b/drivers/misc/ipts/ipts-hid.c
index e85844dc1..53fef2e60 100644
--- a/drivers/misc/ipts/ipts-hid.c
+++ b/drivers/misc/ipts/ipts-hid.c
@@ -350,7 +350,7 @@ static int handle_outputs(ipts_info_t *ipts, int parallel_idx)
 	ipts_buffer_info_t *output_buf, *fb_buf = NULL;
 	u8 *input_report, *payload;
 	u32 transaction_id;
-	int i, payload_size, ret = 0, header_size;
+	int i, payload_size, header_size;
 
 	header_size = sizeof(kernel_output_buffer_header_t);
 	output_buf = ipts_get_output_buffers_by_parallel_id(ipts, parallel_idx);
@@ -415,11 +415,19 @@ static int handle_outputs(ipts_info_t *ipts, int parallel_idx)
 		}
 	}
 
-	if (fb_buf) {
-		ret = ipts_send_feedback(ipts, parallel_idx, transaction_id);
-		if (ret)
-			return ret;
-	}
+	/*
+	 * ipts_send_feedback() causes touch inputs to crash on some
+	 * platforms, especially on Surface Pro 4 and Surface Book 1.
+	 * Let's comment out these lines for now.
+	 *
+	 * Touch and pen issue persists · Issue #374 · jakeday/linux-surface
+	 * https://github.com/jakeday/linux-surface/issues/374#issuecomment-508234110
+	 */
+	// if (fb_buf) {
+	// 	ret = ipts_send_feedback(ipts, parallel_idx, transaction_id);
+	// 	if (ret)
+	// 		return ret;
+	// }
 
 	return 0;
 }
-- 
2.22.0

